interface/src/lib.rs

```diff
@@ -346,6 +347,72 @@ pub enum GetUtxosError {
     MalformedPage { err: String },
 }
 
+/// A request for getting the block headers from a given height.
+#[derive(CandidType, Debug, Deserialize, PartialEq, Eq)]
+pub struct GetBlockHeadersRequest {
+    pub start_height: Height,
+    pub end_height: Option<Height>,
+}
+
+/// The response returned for a request for getting the block headers from a given height.
+#[derive(CandidType, Debug, Deserialize, PartialEq, Eq, Clone)]
+pub struct GetBlockHeadersResponse {
+    pub tip_height: Height,
+    pub block_headers: Vec<BlockHeader>,
+}
+
+/// Errors when processing a `get_block_headers` request.
+#[derive(CandidType, Debug, Deserialize, PartialEq, Eq, Clone)]
+pub enum GetBlockHeadersError {
```

Comment:
Strictly speaking, this type is not part of the interface as it we don't return it in any of our endpoints and it isn't defined in the `.did` file, right? If yes, it would make sense to move this inside the `canister` crate as it'd be considered an implementation detail.

Reply:
That is a valid approach. But we currently have `GetUtxosError`, `SendTransactionError`, and `GetBalanceError` to which all of the things you mentioned apply and it is still in `interface/src/lib.rs`. I think we should keep it consistent, so how about we keep `GetBlockHeadersError ` as is in this PR, and if you still think we should move it to canister, in follow-up I can move all types from above to canister?